mockwebserver/src/main/kotlin/mockwebserver3/MockWebServer.kt

```diff
@@ -232,7 +234,7 @@ class MockWebServer : Closeable {
    * Serve requests with HTTPS rather than otherwise.
    */
   fun useHttps(sslSocketFactory: SSLSocketFactory) {
-    this.sslSocketFactory = sslSocketFactory
+    this.sslSocketFactory = RealOkioSslSocketFactory(sslSocketFactory)
```

Comment:
is it worth us having a more opinionated extenion method for these?

`fun SslSocketFactory.asOkioSocketFactory(): OkioSslSocketFactory = ...`